### PR TITLE
Dev feature message

### PR DESCRIPTION
**Pull request by cheebo**

### CODE REVIEW

File: common.proto [hunk 1]
```diff
@@ -4,11 +4,11 @@ package proto;
 option go_package = "pkg/proto";
 
 message ID {
```
Comment (bruteforce1414) [inline common.proto:6]: ID isn't CamelCase

File: common.proto [hunk 2]
```diff
@@ -4,11 +4,11 @@ package proto;
...
-    string Id = 1;
-    string Version=2;
+    string id = 1;
```
Comment (bruteforce1414) [inline common.proto]: It seems strange that Id consists of id.
Maybe Id will consist of name instead id?

File: config.proto [hunk 3]
```diff
@@ -5,34 +5,31 @@ option go_package = "pkg/proto";
...
-//config set
+// cmd: config set
 message ConfigSetRequest{
-    ID Id=1;
-    string Key=2;
-    string Value=3;
+    ID id = 1;
+    string key = 2;
+    string value = 3;
 }
```
Comment (bruteforce1414) [inline config.proto:22]: add indent between rows?

File: config.proto [hunk 4]
```diff
@@ -5,34 +5,31 @@ option go_package = "pkg/proto";
...
 message ConfigSetReply{
-    ErrorReply ErrorConfig=1;
+    Error error = 1;
 }
 
-//config upload
+// cmd: config upload
 message ConfigUploadRequest{
-    string Path=1;
+    string path = 1;
```
Comment (bruteforce1414) [inline config.proto]: change type to bytes?

File: meta.proto [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+message Meta{
+    ID id = 1;
+    Author author = 2;
+    repeated Dependency dependencies = 3;
+    Description description = 4;
+    Core core = 5;
+    string interface = 6;
+    repeated License licenses = 7;
+    repeated Link links = 8;
+    Repository repository = 9;
```
Comment (bruteforce1414) [inline meta.proto:17]: repository->repositories

File: meta.proto [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+message Core {
+    string version_constraint = 1;
+}
+
+message License {
+    string title = 1;
+    string type = 2;
+    string uri = 3;
```
Comment (bruteforce1414) [inline meta.proto:43]: all URL are URI, and maybe will change all variable "uri" to url for simplification

File: nori.proto [hunk 7]
```diff
@@ -8,19 +8,18 @@ import "config.proto";
...
-    rpc PluginStartCommand (PluginStartRequest)returns (ErrorReply){}
-    rpc PluginStopCommand (PluginStopRequest)returns (ErrorReply){}
-    rpc PluginUninstallCommand (PluginUninstallRequest) returns (ErrorReply) {}
-    rpc PluginUploadCommand (PluginUploadRequest) returns (ErrorReply) {}
+    rpc PluginPullCommand (PluginPullRequest) returns (Error) {}
+    rpc PluginRemoveCommand (PluginRemoveRequest) returns (Error) {}
+    rpc PluginStartCommand (PluginStartRequest)returns (Error){}
+    rpc PluginStopCommand (PluginStopRequest)returns (Error){}
+    rpc PluginUninstallCommand (PluginUninstallRequest) returns (Error) {}
+    rpc PluginUploadCommand (stream PluginUploadChunk) returns (Error) {}
```
Comment (bruteforce1414) [inline nori.proto:22]: define in config.proto PluginUploadChunk

File: plugin.proto [hunk 8]
```diff
@@ -1,116 +1,99 @@
...
 
 // cmd: plugin list
 message PluginListRequest{
-    bool FlagAll=1;
-    bool FlagError=2;
-    bool FlagInstalled = 3;
-    bool FlagRunning = 4;
-    bool FlagInstallable=5;
-    bool FlagInactive=6;
+    bool flag_all = 1;
```
Comment (bruteforce1414) [inline plugin.proto:43]: Maybe add flag_stopped, flag_inited, flag_active?

File: plugin.proto [hunk 9]
```diff
@@ -1,116 +1,99 @@
...
-    bool FlagRunning = 14;
-    bool FlagInstallable=15;
-    bool FlagInactive=16;
+// cmd: plugin meta
+message PluginMetaRequest {
+    ID id = 1;
+    bool flag_deps = 2;
+    bool flag_deps_status = 3;
+    bool flag_dependent = 4;
+    bool flag_dependent_status = 5;
```
Comment (bruteforce1414) [inline plugin.proto:57]: flag_dependent->flag_dependent_of_this_plugin to simplification
?

**Review by bruteforce1414 — COMMENTED** [review]
There are some ideas to change code

File: meta.proto (hunk 5 above)
Comment (cheebo) [reply meta.proto:17]: no, we have only one repository.

File: nori.proto (hunk 7 above)
Comment (cheebo) [reply nori.proto:22]: why in config.proto when it's plugin command?

File: plugin.proto (hunk 9 above)
Comment (cheebo) [reply plugin.proto:57]: I believe "flag_dependent" good enough. it's short and obvious

**Pull request merged**